DigitalOffice/Services/UserService/Commands/DisableUserByIdCommand.cs

```diff
@@ -0,0 +1,59 @@
+﻿using LT.DigitalOffice.Broker.Requests;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.UserService.Commands.Interfaces;
+using LT.DigitalOffice.UserService.Database.Entities;
+using LT.DigitalOffice.UserService.Repositories.Interfaces;
+using MassTransit;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.UserService.Commands
+{
+    public class DisableUserByIdCommand : IDisableUserByIdCommand
+    {
+        private readonly IUserRepository repository;
+        private readonly IRequestClient<IUserIdRequest> requestClient;
+
+        public DisableUserByIdCommand(
+            [FromServices] IUserRepository repository,
+            [FromServices] IRequestClient<IUserIdRequest> requestClient)
+        {
+            this.repository = repository;
+            this.requestClient = requestClient;
+        }
+
+        public async Task<bool> ExecuteAsync(Guid userId, Guid requestingUserId)
+        {
+            var isUserRight = await GetResultCheckingUserRights(requestingUserId);
+
+            if (!isUserRight)
+            {
+                throw new Exception("Not enough rights");
+            }
+
+            DbUser editedUser = repository.GetUserInfoById(userId);
+
+            editedUser.IsActive = false;
+
+            return repository.EditUser(editedUser);
+        }
+
+        private async Task<bool> GetResultCheckingUserRights(Guid userId)
+        {
```

Comment:
Please w8 for AccessValidator in develop and use it

Reply:
Fixed